stackslib/src/net/api/gettenureblocks.rs

```diff
@@ -0,0 +1,257 @@
+    }
+
+    fn metrics_identifier(&self) -> &str {
+        "/v3/tenures/blocks/:consensus_hash"
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        captures: &Captures,
+        query: Option<&str>,
+        _body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() != 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected 0-length body".to_string(),
+            ));
+        }
+        let consensus_hash = request::get_consensus_hash(captures, "consensus_hash")?;
+        self.consensus_hash = Some(consensus_hash);
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCNakamotoTenureBlocksRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.consensus_hash = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let consensus_hash = self
+            .consensus_hash
+            .take()
+            .ok_or(NetError::SendError("`consensus_hash` not set".into()))?;
+
+        let tenure_blocks_resp =
+            node.with_node_state(|_network, sortdb, chainstate, _mempool, _rpc_args| {
+                let header_info =
+                    match NakamotoChainState::find_highest_known_block_header_in_tenure(
+                        &chainstate,
+                        sortdb,
+                        &consensus_hash,
+                    ) {
+                        Ok(Some(header)) => header,
+                        Ok(None) => {
+                            let msg = format!("No blocks in tenure {}", &consensus_hash);
+                            debug!("{}", &msg);
+                            return Err(StacksHttpResponse::new_error(
+                                &preamble,
+                                &HttpNotFound::new(msg),
+                            ));
+                        }
+                        Err(e) => {
+                            let msg = format!(
+                                "Failed to query tenure blocks by consensus '{}': {:?}",
+                                consensus_hash, &e
+                            );
+                            error!("{}", &msg);
+                            return Err(StacksHttpResponse::new_error(
+                                &preamble,
+                                &HttpServerError::new(msg),
+                            ));
+                        }
+                    };
+
+                let blocks = match NakamotoChainState::get_block_headers_in_tenure_at_burnview(
```

Comment:
Given that this is going to be an I/O-intensive function (_O(n log n)_ database reads for an _n_-block tenure), it would be better to break up the code body into an `impl HttpChunkGenerator`, which processes one tenure block at a time. Then, the HTTP server will call it once per I/O operation, so this RPC call won't starve the rest of the networking stack.  For example, see how `getblocks_v3.rs` does it.